Blockly Factory: New Selector UI

quachtina96 opened the pull request:
## Users may now scroll through their block library blocks and select their blocks by clicking on a block option (the label, checkbox, or preview workspace). Checkboxes have been added to clearly convey that blocks are being selected for export.
<img width="751" alt="screen shot 2016-08-22 at 5 04 14 pm" src="https://cloud.githubusercontent.com/assets/10423718/17875868/7c06de00-688a-11e6-897e-3f9b00d00dd8.png">


## Code Summary
- created BlockOption class 
- added function to generate the selector from the block library in ExporterTools
- changed BlockExporterController & BlockExporterView to accommodate use of BlockOption
- added styles for BlockOption and selector
- moved resize out of AppController init function and recenter block options in exporter upon window resize.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly/579)
<!-- Reviewable:end -->

quachtina96 commented:
@evd2014 @vicng @picklesrus

vicng commented:
Nice work (and in a short amount of time too)! The UI is a lot more intuitive now.

Comments below.

---

Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/app_controller.js, line 499 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPooJT8MtF7CVQ2AlZO:-KPooJT8MtF7CVQ2AlZP:b-h77l5t) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/app_controller.js#L499)):*
> ```JavaScript
>   }
> 
>   // Assign click handlers.
> ```

Sorry I didn't notice this before (it's been hard with so many different CLs), but the responsibility of AppController is confusing. I'd imagine that it should only be responsible for tab management and for coordinating logic for any shared data. However, it is also a controller of other functions within the BlockFactory tab and Exporter tabs (those should be handled by their own controllers).

You don't have to do this in this CL, but if you have time later, try refactoring AppController (eg. the assignment of these click handlers belong in BlockExporterController).

---

*[demos/blocklyfactory/block_exporter_controller.js, line 169 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorQNqIjeGW7wcjC6w:-KPorQNr_WbeVWTexGmb:b-svz9cw) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L169)):*
> ```JavaScript
>  * Adds all blocks stored in block library to the selector workspace.
>  */
> BlockExporterController.prototype.addAllBlocksToWorkspace = function() {
> ```

Is there a workspace now?

---

*[demos/blocklyfactory/block_exporter_controller.js, line 197 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorkFImUZouywINoiO:-KPorkFJGRynoCDl2Yda:bezm8em) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L197)):*
> ```JavaScript
>     // Use an additional closure to correctly assign the tab callback.
>     blockOption.dom.addEventListener(
>         'click', self.makeBlockOptSelectHandler_(blockOption));
> ```

If this is the only method that will be using it, can makeBlockOptSelectHandler_ and updateSelectedBlockTypes_ be defined inline here?

---

*[demos/blocklyfactory/block_exporter_view.js, line 128 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPotz74YRa7dpzmexl6:-KPotz74YRa7dpzmexl7:bmxyzyc) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_view.js#L128)):*
> ```JavaScript
>  */
> BlockExporterView.prototype.setSelectedBlockTypes = function(blockTypes) {
>   for (var i =0, blockType; blockType = blockTypes[i]; i++) {
> ```

Nit: i = 0 

---

*[demos/blocklyfactory/block_option.js, line 138 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPovmoyiNSzI0ZWsi03:-KPovmozDZngQCspaRrl:bt0zjyn) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L138)):*
> ```JavaScript
>  * Centers the preview block in the workspace.
>  */
> BlockOption.prototype.centerBlock = function() {
> ```

It'd be nice if you could have the workspace sized to fit the block. Don't need to do it in this CL, but later on if you have time.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPosYovwCbnddu0xTXN:bhe6y58)*
<!-- Sent from Reviewable.io -->

picklesrus commented:
+1 to vicng's comments and a few nits sprinkled in there.  I like this!

---

Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 53 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorpveHYlMugb6NTSI:-KPorpvfnv8W2zsBQgE4:b-hwxdgs) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L53)):*
> ```JavaScript
>   // Map of block types stored in block library to their corresponding Block
>   // Option objects.
>   this.blockOptions = this.tools.createBlockSelectorFromLib(this.blockLibStorage,
> ```

nit: > 80 col

---

*[demos/blocklyfactory/block_exporter_view.js, line 64 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPow_2C8rQ9vuPHT_Uz:-KPow_2C8rQ9vuPHT_V-:by97rh4) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_view.js#L64)):*
> ```JavaScript
>  * @param {!Object} blockOptions - Map of block types to BlockOption objects.
>  */
> BlockExporterView.prototype.getBlockOptions = function(blockOpts) {
> ```

is this called? Can't seem to find where.  And looks more like a setter than a getter :)

---

*[demos/blocklyfactory/block_option.js, line 66 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPowtzInZRMsisIwnkf:-KPowtzJlbJxjS92qLXg:bb6p57j) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L66)):*
> 80

---

*[demos/blocklyfactory/block_option.js, line 76 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPowvm6gfq4bUK5y-BP:-KPowvm6gfq4bUK5y-BQ:b-rs03fz) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L76)):*
> ```JavaScript
>   }, ''); // Empty quotes for empty div.
> 
>   // Create and append div in which to inject the workspace for viewing the block
> ```

also > 80?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPoxEKCSAk1z4zdisQf:b-m54ot1)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/app_controller.js, line 499 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPooJT8MtF7CVQ2AlZO:-KPowPzQOur_3QGe53fb:bz2uvxl) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/app_controller.js#L499)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> Sorry I didn't notice this before (it's been hard with so many different CLs), but the responsibility of AppController is confusing. I'd imagine that it should only be responsible for tab management and for coordinating logic for any shared data. However, it is also a controller of other functions within the BlockFactory tab and Exporter tabs (those should be handled by their own controllers).
> 
> You don't have to do this in this CL, but if you have time later, try refactoring AppController (eg. the assignment of these click handlers belong in BlockExporterController).
> 
</details>
Yes, agreed. I've been wanting to refactor appcontroller but would like to make sure everything is working before I do that :)

---

*[demos/blocklyfactory/block_exporter_controller.js, line 53 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorpveHYlMugb6NTSI:-KPozSnxj27hDAf9JeIP:b-896fix) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L53)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> nit: > 80 col
</details>
Done.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 169 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorQNqIjeGW7wcjC6w:-KPow_XcGoi0mj1SDukX:bg05r04) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L169)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> Is there a workspace now?
</details>
oops renamed! and I also removed those from the comments.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 197 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPorkFImUZouywINoiO:-KPoz7tDoqnW4ry4gxnc:b-896fix) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_controller.js#L197)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> If this is the only method that will be using it, can makeBlockOptSelectHandler_ and updateSelectedBlockTypes_ be defined inline here?
</details>
Done.

---

*[demos/blocklyfactory/block_exporter_view.js, line 64 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPow_2C8rQ9vuPHT_Uz:-KPozbQLWrZj8k7XJq-a:bweflkq) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_view.js#L64)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> is this called? Can't seem to find where.  And looks more like a setter than a getter :)
</details>
i removed this whole function!

---

*[demos/blocklyfactory/block_exporter_view.js, line 128 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPotz74YRa7dpzmexl6:-KPoziA5ZkUT7CPzSx40:b-896fix) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_exporter_view.js#L128)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> Nit: i = 0 
</details>
Done.

---

*[demos/blocklyfactory/block_option.js, line 66 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPowtzInZRMsisIwnkf:-KPozsIOPW6MZLyo9atx:b-896fix) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L66)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> > 80
</details>
Done.

---

*[demos/blocklyfactory/block_option.js, line 76 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPowvm6gfq4bUK5y-BP:-KPozsjYp5pl4402Yozl:b-896fix) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L76)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> also > 80?
</details>
Done.

---

*[demos/blocklyfactory/block_option.js, line 138 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPovmoyiNSzI0ZWsi03:-KPp-0gXBsUn6AcMCHCo:bie683w) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L138)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> It'd be nice if you could have the workspace sized to fit the block. Don't need to do it in this CL, but later on if you have time.
</details>
What do you mean by this? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
I just rebased in order to get rid of merge conflicts

---

Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPsJ2b3CClnFlbO183A:b9ukyql)*
<!-- Sent from Reviewable.io -->

evd2014 commented:
This looks really great! The selector looks great, and nice work with the BlockOption class.

In a later CL maybe you can play around with padding and margins to get the divs all centered in the BlockSelector column so there's an equal amount of gray on either side? That's pretty low priority though. 

---

Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 50 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsE1FdKaYXpEQiFFYF:-KPsE1FdKaYXpEQiFFYG:b-cz0ch8) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L50)):*
> ```JavaScript
>   // Utils for generating code to export.
>   this.tools = new BlockExporterTools();
>   this.selectorID = 'blockSelector';
> ```

Comment for selectorID? Took me a while to figure out what it was. 

---

*[demos/blocklyfactory/block_exporter_controller.js, line 87 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsEFjM4pM6BxX7-TKQ:-KPsEFjM4pM6BxX7-TKR:bo3bbey) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L87)):*
> ```JavaScript
>  * @return {!Array.<string>} Types of blocks selected.
>  */
> BlockExporterController.prototype.getSelectedBlockTypes_ = function() {
> ```

You only call this method in one place (export) and it seems to be a private method that just makes a method call on an instance variable. This would make sense if it was being called externally, but as it seems to be a private function, remove this function entirely and just call this.view.getSelectedBlockTypes() in export().

---

*[demos/blocklyfactory/block_exporter_controller.js, line 200 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsEqSNmaisXhPdf5wP:-KPsEqSO7Q40Hx7MIbBR:b-lytp0q) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L200)):*
> ```JavaScript
>   var updateSelectedBlockTypes_ = function(blockOption) {
>     // Toggle selected.
>     var selected = blockOption.isSelected() ? false : true;
> ```

Replace with blockOption.setSelected(!blockOption.isSelected())

---

*[demos/blocklyfactory/block_exporter_view.js, line 132 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsLR7RJfmKBiWof0R-:-KPsLR7SQ2DLZwG1FWCY:b-exmfht) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_view.js#L132)):*
> ```JavaScript
>     var blockOption = this.blockOptions[blockType];
>     if (blockOption.isSelected()) {
>       selectedTypes.push(blockOption.blockType);
> ```

Is blockOption.blockType the same as blockType? If so, it would be more readable to push blockType instead of blockOption.blockType. If they're different, consider using a different variable name when iterating through blockOptions.

---

*[demos/blocklyfactory/block_option.js, line 59 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsLprkICHfo5bXs0E4:-KPsLprkICHfo5bXs0E5:bg1rwl) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_option.js#L59)):*
> ```JavaScript
>   this.previewWorkspace = null;
>   // Whether or not block the option is selected.
>   this.selected = false;
> ```

Is there a reason to have a selected state variable in addition to the checkbox? The selection information should also be stored in this.checkbox.checked, and also having a selected variable creates the possibility for the two to get out of sync. Because that information is tied to use input, I think it would be OK to have the state stored in the checkbox. The only things you would need to update are isSelected() and setSelected(). 

---

*[demos/blocklyfactory/factory.css, line 218 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNhwJA5qpJNruKvQe:-KPsNhwJA5qpJNruKvQf:b-lzp9vv) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L218)):*
> ```CSS
> #blockSelector {
>   background-color: lightgrey;
>   width: 80%;
> ```

Nit alphabetize

---

*[demos/blocklyfactory/factory.css, line 235 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNkT5lgQr9S7ptEIB:-KPsNkT5lgQr9S7ptEIC:b-5i3q5n) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L235)):*
> ```CSS
> .blockOption_check_label {
>   position: relative;
>   display: none;
> ```

nit alphabetize

---

*[demos/blocklyfactory/factory.css, line 247 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNlzxXojspgVnrQZZ:-KPsNlzxXojspgVnrQZ_:b-5i3q5n) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L247)):*
> ```CSS
>     max-width: inherit;
>     word-wrap: break-word;
>     overflow-y: scroll;
> ```

nit alphabetize

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPsNyAcaluKrd2ZDOWt:b-adw449)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 50 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsE1FdKaYXpEQiFFYF:-KPsUSNWvOBdKIkwp8MO:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L50)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> Comment for selectorID? Took me a while to figure out what it was. 
</details>
Done.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 87 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsEFjM4pM6BxX7-TKQ:-KPsW4pf5N8QCjOn2F6X:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L87)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> You only call this method in one place (export) and it seems to be a private method that just makes a method call on an instance variable. This would make sense if it was being called externally, but as it seems to be a private function, remove this function entirely and just call this.view.getSelectedBlockTypes() in export().
</details>
Done.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 200 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsEqSNmaisXhPdf5wP:-KPsWClA7Rl3VjE6PY7E:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_controller.js#L200)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> Replace with blockOption.setSelected(!blockOption.isSelected())
</details>
Done.

---

*[demos/blocklyfactory/block_exporter_view.js, line 132 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsLR7RJfmKBiWof0R-:-KPsWaEN-puznSw7qe6S:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_exporter_view.js#L132)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> Is blockOption.blockType the same as blockType? If so, it would be more readable to push blockType instead of blockOption.blockType. If they're different, consider using a different variable name when iterating through blockOptions.
</details>
Done.

---

*[demos/blocklyfactory/block_option.js, line 59 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsLprkICHfo5bXs0E4:-KPsPErh89eR0dK0JJTN:b-ly3o9n) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/block_option.js#L59)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> Is there a reason to have a selected state variable in addition to the checkbox? The selection information should also be stored in this.checkbox.checked, and also having a selected variable creates the possibility for the two to get out of sync. Because that information is tied to use input, I think it would be OK to have the state stored in the checkbox. The only things you would need to update are isSelected() and setSelected(). 
</details>
I originally was maintaining state only through the checkboxes but this caused trouble in handling the click. The goal is that when the user clicks on the block option (on the div containing the label, checkbox, and preview), the block is selected. Without maintaining this state, clicking on the checkbox (and toggling selected) doesn't select/deselect the block.

---

*[demos/blocklyfactory/factory.css, line 218 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNhwJA5qpJNruKvQe:-KPsWLIgnh8c8B2Wf80L:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L218)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> Nit alphabetize
</details>
Done.

---

*[demos/blocklyfactory/factory.css, line 235 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNkT5lgQr9S7ptEIB:-KPsPvWmQNvlpKhI18ji:b-t2iswb) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L235)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> nit alphabetize
</details>
removed display:none

---

*[demos/blocklyfactory/factory.css, line 247 \[r2\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsNlzxXojspgVnrQZZ:-KPsQ1zKML_CKsyVqche:b-896fix) ([raw file](https://github.com/google/blockly/blob/329a65c09251421b6b5f6618cee0b8ba2dfabafe/demos/blocklyfactory/factory.css#L247)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> nit alphabetize
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579)*
<!-- Sent from Reviewable.io -->

picklesrus commented:
Review status: 0 of 7 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 216 \[r4\]](https://reviewable.io:443/reviews/google/blockly/579#-KPswIVFD9hewYXEgjf5:-KPswIVGXj3aivJTEu24:b-cyluj2) ([raw file](https://github.com/google/blockly/blob/adbc08f738ddfc35f261680a55854a09df7cf7f3/demos/blocklyfactory/block_exporter_controller.js#L216)):*
> ```JavaScript
>       updateSelectedBlockTypes_(blockOption);
>       // TODO(quachtiana96): uncomment line below once CL is merged.
>       // self.updatePreview();
> ```

That CL is merged now right? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
Review status: 0 of 7 files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 216 \[r4\]](https://reviewable.io:443/reviews/google/blockly/579#-KPswIVFD9hewYXEgjf5:-KPsz-RyoY3jdDbonkGW:b-pjxbdw) ([raw file](https://github.com/google/blockly/blob/adbc08f738ddfc35f261680a55854a09df7cf7f3/demos/blocklyfactory/block_exporter_controller.js#L216)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> That CL is merged now right? 
</details>
Yep! I rebased and also added a few more small changes in my latest revision

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
@vicng @evd2014 @picklerus I've addressed all outstanding comments. Updated UI is below:

![Screen Shot 2016-08-23 at 1.00.35 PM.png](https://files.reviewableusercontent.io/images/reviewable/10423718/IMsIBlf0E0vCZ0PqpHMWh2HN/Screen%20Shot%202016-08-23%20at%201.00.35%20PM.png)

---

Review status: 0 of 7 files reviewed at latest revision, 16 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPszRczgoJaRR9QTVlD:b-nrrn7o)*
<!-- Sent from Reviewable.io -->

vicng commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 188 \[r4\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsxvrA2UGp9XYvkim1:-KPsxvrA2UGp9XYvkim2:btsja5y) ([raw file](https://github.com/google/blockly/blob/adbc08f738ddfc35f261680a55854a09df7cf7f3/demos/blocklyfactory/block_exporter_controller.js#L188)):*
> ```JavaScript
> BlockExporterController.prototype.addBlockOptSelectHandlers = function() {
>   var self = this;
>   /**
> ```

You can remove the JSDoc here and just add a quick comment about what each function does.

---

*[demos/blocklyfactory/block_option.js, line 138 \[r1\]](https://reviewable.io:443/reviews/google/blockly/579#-KPovmoyiNSzI0ZWsi03:-KPszRVH5zcjHIZi0-Xf:b-ophh1s) ([raw file](https://github.com/google/blockly/blob/71b30f8f1ef10cfa6afbcce2dc3cd92ee35a19d2/demos/blocklyfactory/block_option.js#L138)):*
<details><summary><i>Previously, quachtina96 wrote…</i></summary>
> What do you mean by this? 
</details>
The workspace area for each block looks like it's fixed (100px height). It'd be nice if each workspace div was just large enough to render the block it holds, resulting in a more svelte list :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPt0xkFK1BmEHCFdcVs:b-uwkowd)*
<!-- Sent from Reviewable.io -->

picklesrus commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579#-:-KPt2O7LZ6VlGR2Otx9k:bnfp4nl)*
<!-- Sent from Reviewable.io -->

quachtina96 commented:
Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 188 \[r4\]](https://reviewable.io:443/reviews/google/blockly/579#-KPsxvrA2UGp9XYvkim1:-KPt3igQtreugNXeGHWS:b-896fix) ([raw file](https://github.com/google/blockly/blob/adbc08f738ddfc35f261680a55854a09df7cf7f3/demos/blocklyfactory/block_exporter_controller.js#L188)):*
<details><summary><i>Previously, vicng wrote…</i></summary>
> You can remove the JSDoc here and just add a quick comment about what each function does.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/579)*
<!-- Sent from Reviewable.io -->
